libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxOfflineRouter.java

```diff
@@ -6,23 +6,38 @@
 import com.mapbox.api.directions.v5.models.DirectionsRoute;
 import com.mapbox.navigator.Navigator;
 
+import java.io.File;
+
 public class MapboxOfflineRouter {
+  private final String TILE_PATH_NAME = "tiles";
   private final OfflineNavigator offlineNavigator;
+  private final String tilePath;
+
+  /**
+   * Creates an offline router which uses the specified offline path for storing and retrieving
+   * data.
+   *
+   * @param offlinePath directory path where the offline data is located
+   */
+  public MapboxOfflineRouter(String offlinePath) {
+    File tileDir = new File(offlinePath, TILE_PATH_NAME);
+    if (!tileDir.exists()) {
+      tileDir.mkdirs();
+    }
 
-  public MapboxOfflineRouter() {
+    this.tilePath = tileDir.getAbsolutePath();
     offlineNavigator = new OfflineNavigator(new Navigator());
   }
 
   /**
-   * Configures the navigator for getting offline routes
+   * Configures the navigator for getting offline routes.
    *
-   * @param tilesDirPath directory path where the tiles are located
    * @param callback a callback that will be fired when the offline data is initialized and
-   * {@link MapboxOfflineRouter#findOfflineRoute(OfflineRoute, OfflineRouteFoundCallback)}
+   * {@link MapboxOfflineRouter#findOfflineRoute(OfflineRoute, RouteFoundCallback)}
    *                 can be called safely.
    */
-  public void initializeOfflineData(String tilesDirPath, OnOfflineDataInitialized callback) {
-    offlineNavigator.configure(tilesDirPath, callback);
+  public void initializeOfflineData(String version, OnOfflineDataInitialized callback) {
```

Comment:
@Guardiola31337 I actually think `configureRouter` would make the most sense here, because you are configuring it be ready to use the version you specified. Do we know why we went with `initializeData`?